Updated autoprefixer call for version >= 5.2

netsgnut opened the pull request:
Starting from `autoprefixer-core` 5.2, the `ap.process()` is obsolete (Ref: postcss/autoprefixer-core@d35c9321e2ca956fafc82e443e3b2cd06c21aaab). This PR aims to fix this by updating the call to the newer recommended way.

Please advise. Thanks.

puzrin commented:
Thanks for PR! It's ok in general, but i think it's not save to rely on `postcss` availability in version check.

I could merge PR as useful enougth, but it needs more fixes for release.

- more safe version check
- probably, the same fixes in `css_wring` plugins

netsgnut commented:
Thanks for the feedback! In our case here, it seems better to sniff for postcss 4.1 instead of 4.0, using `postcss.plugin`. I have updated the PR to address this. Also updated is the `CsswringCompressor` to cope with the upcoming API change introduced by hail2u/node-csswring@4921ad316bd72aea5428fdc3d799322465923e93.

puzrin commented:
Looks good. I'll bump version to 1.3.0 for sure.

@ixti , take a look please.

ixti commented on lib/mincer/compressors/csswring_compressor.js:
```diff
@@ -43,15 +48,25 @@ require('util').inherits(CsswringCompressor, Template);
+
+    return csswring.wring(css, config);
+  };
+
```
This should go to constructor - it needs to be initialized only once not upon each file evaluation.

ixti commented:
LGTM. Ok to be merged once `wring` function initialization will be [moved out](https://github.com/nodeca/mincer/pull/200#discussion_r32620479) of `evaluate`

netsgnut commented:
Sure, I will update the PR right away to move that to initialiser.

netsgnut commented:
A bit OT, but as working on the PR, it seems that there are dependencies that are specified by `'*'`, but actually have minimum working version in the actual implementation, like `sass` (requires >= 0.5) and `uglify` (requires > 1). Do you think it is a good idea to update the dependencies in `package.json`, while releasing the 1.3.0 build? Thanks.

puzrin commented:
@netsgnut i don't see reasons to overcomplicate. Those deps are for developpers only anyway.

netsgnut commented:
@puzrin Indeed

ixti commented:
@netsgnut Can you squash it to a single commit, please?

netsgnut commented:
@ixti Sure, rebased as f8e0906.

ixti commented:
Thanks!

puzrin commented:
Released with you commit. Thanks!

netsgnut commented:
Saw that on NPM as well. Thanks! <3
